src/main/kotlin/com/sourcegraph/cody/history/HistoryService.kt

```diff
@@ -93,6 +90,16 @@ class HistoryService(private val project: Project) :
                 Speaker.HUMAN -> MessageState.SpeakerState.HUMAN
                 Speaker.ASSISTANT -> MessageState.SpeakerState.ASSISTANT
               }
+          val contextFiles = chatMessage.contextFiles
+          if (!contextFiles.isNullOrEmpty()) {
+            contextFiles.forEach { contextFile ->
+              val element = ContextFile()
+              element.uri = contextFile.uri.toString()
+              if (!it.contextFiles.contains(element)) {
```

Comment:
In general the approach is good. However I think we should be more precise about the saved files. 

1. Probably we should distinguish the local files and the files from the remote repo contexts.
2. We should consider persisting a given file only once - it this implementation the same file but different fragment (the lines, see example below). Maybe we should save only the files that are presented? 🤔 

Note that VS Code persists the context files with their proper links. 

```
{
  "uri": "https://sourcegraph.sourcegraph.com//github.com/sourcegraph/jetbrains@3761a12f72c5bcf8f777924d75f32db729c57788/-/blob/CONTRIBUTING.md?L49-56",
  "repoName": "github.com/sourcegraph/jetbrains",
  "revision": "3761a12f72c5bcf8f777924d75f32db729c57788",
  "range": {
    "start": {
      "line": 48,
      "character": 0
    },
    "end": {
      "line": 56,
      "character": 0
    }
  },
  "type": "file"
},
```

Reply:
Why do we have this code *at all* in JetBrains, instead of relying on the extension code's history? The extension shards chat history by account, preventing bugs like this, #1210 that I noticed yesterday.